draft-ietf-quic-transport.md

```diff
@@ -1461,127 +1461,126 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+
+PATH_CHALLENGE ({{frame-path-challenge}}) and PATH_RESPONSE
+({{frame-path-response}}) frames MAY be used for confirming that an endpoint
+owns an address, or to verify the PMTU to/from an address.
+
+An endpoint validates a peer's address by sending a PATH_CHALLENGE frame
+containing a payload that is hard to guess.  This frame MUST be sent in a packet
+to the new address.  Once a PATH_RESPONSE frame containing the same payload is
+received from the peer, the address is considered to be valid.  A PATH_CHALLENGE
+frame MUST contain at least 12 randomly generated octets {{?RFC4086}}, making
+the payload sufficiently hard to guess.
+
+An endpoint SHOULD not generate two PATH_CHALLENGE frames with the same payload.
+
+
+### Client Behavior {{migration-client}}
+
+When a new local address is available, a client MUST first send a packet
+containing a PATH_CHALLENGE frame from this new address before sending any other
+packets.  The PATH_CHALLENGE frame MAY be bundled with other frames, but this
+packet MUST be padded to exactly 1200 octets unless the client has a reasonable
+assurance that the PMTU is larger.  Sending a packet of this size confirms that
+the network path from the client to the server supports an MTU of this size and
+helps reduce the amplitude of amplification attacks caused by server responses
+toward an unverified client address.
+
+A client MAY send additional PATH_CHALLENGE frames to handle packet loss or to
+get more measurements on the new network path, subject to the congestion
+controller's limits. The client SHOULD use fresh random data in each
+PATH_CHALLENGE frame so that it can associate the server's response with the
+causative PATH_CHALLENGE.
+
+When the client receives a PATH_RESPONSE from the server and verifies that the
+included data matches that sent by the client in a previous PATH_CHALLENGE, the
+server is considered reachable from the new address and the PMTU of the new
+network path is considered verified.
+
+A client SHOULD abandon the new address when it does not receive a PATH_RESPONSE
+after sending some number of PATH_CHALLENGE frames and/or after some time has
+passed. Note that the client may be receiving data from the server during this
+period, but not receiving a PATH_RESPONSE may be indicative of a PMTU detection
+failure.
+
+A client may receive a PATH challenge from a server that is validating the
+client's ownership of an address. A client responds to this frame by echoing the
+data from the PATH_CHALLENGE frame in a PATH_RESPONSE frame.  This response MAY
+be bundled with other frames and does not require additional padding.
+
+
+### Server Behavior {{migration-server}}
+
+When a server receives a PATH_CHALLENGE frame from a client, it MUST respond by
+copying the included data in a PATH_RESPONSE frame, and send it in a packet
+padded to the size of the packet carrying the PATH_CHALLENGE. This PATH_RESPONSE
+MUST be sent to the address that the PATH_CHALLENGE was received from.
```

Comment:
Minor wordsmithing:

When a server receives a PATH_CHALLENGE frame from a client, it MUST generate a PATH_RESPONSE frame that contains the same data as the challenge, and send it in a packet padded to the size of the client's packet that included the PATH_CHALLENGE. This PATH_RESPONSE MUST be sent to the address from which the PATH_CHALLENGE was received.

Reply:
Done